zetaclient/orchestrator/bootstrap.go

```diff
@@ -0,0 +1,343 @@
+	"github.com/zeta-chain/zetacore/zetaclient/db"
+	"github.com/zeta-chain/zetacore/zetaclient/metrics"
+)
+
+// backwards compatibility
+const btcDatabaseFilename = "btc_chain_client"
+
+// CreateSignerMap creates a map of interfaces.ChainSigner (by chainID) for all chains in the config.
+// Note that signer construction failure for a chain does not prevent the creation of signers for other chains.
+func CreateSignerMap(
+	ctx context.Context,
+	tss interfaces.TSSSigner,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+) (map[int64]interfaces.ChainSigner, error) {
+	signers := make(map[int64]interfaces.ChainSigner)
+	_, _, err := syncSignerMap(ctx, tss, logger, ts, &signers)
+	if err != nil {
+		return nil, err
+	}
+
+	return signers, nil
+}
+
+// syncSignerMap synchronizes the given signers map with the signers for all chains in the config.
+// This semantic is used to allow dynamic updates to the signers map.
+func syncSignerMap(
+	ctx context.Context,
+	tss interfaces.TSSSigner,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+	signers *map[int64]interfaces.ChainSigner,
+) (int, int, error) {
+	if signers == nil {
+		return 0, 0, errors.New("signers map is nil")
+	}
+
+	app, err := zctx.FromContext(ctx)
+	if err != nil {
+		return 0, 0, errors.Wrapf(err, "failed to get app context")
+	}
+
+	var (
+		added, removed int
+
+		presentChainIDs = make([]int64, 0)
+
+		onAfterSet = func(chainID int64, _ interfaces.ChainSigner) {
+			logger.Std.Info().Msgf("Added signer for chain %d", chainID)
+			added++
+		}
+
+		onBeforeUnset = func(chainID int64, _ interfaces.ChainSigner) {
+			logger.Std.Info().Msgf("Removing signer for chain %d", chainID)
+			removed++
+		}
+	)
+
+	// EVM signers
+	for _, evmConfig := range app.Config().GetAllEVMConfigs() {
+		chainID := evmConfig.Chain.ChainId
+
+		if evmConfig.Chain.IsZetaChain() {
+			continue
+		}
+
+		evmChainParams, found := app.GetEVMChainParams(chainID)
+		switch {
+		case !found:
+			logger.Std.Warn().Msgf("Unable to find chain params for EVM chain %d", chainID)
+			continue
+		case !evmChainParams.IsSupported:
+			logger.Std.Warn().Msgf("EVM chain %d is not supported", chainID)
+			continue
+		}
+
+		presentChainIDs = append(presentChainIDs, chainID)
+
+		// noop for existing signers
+		if mapHas(signers, chainID) {
+			continue
+		}
+
+		var (
+			mpiAddress          = ethcommon.HexToAddress(evmChainParams.ConnectorContractAddress)
+			erc20CustodyAddress = ethcommon.HexToAddress(evmChainParams.Erc20CustodyContractAddress)
+		)
+
+		signer, err := evmsigner.NewSigner(
+			ctx,
+			evmConfig.Chain,
+			tss,
+			ts,
+			logger,
+			evmConfig.Endpoint,
+			config.GetConnectorABI(),
+			config.GetERC20CustodyABI(),
+			mpiAddress,
+			erc20CustodyAddress,
+		)
+		if err != nil {
+			logger.Std.Error().Err(err).Msgf("Unable to construct signer for EVM chain %d", chainID)
+			continue
+		}
+
+		mapSet[int64, interfaces.ChainSigner](signers, chainID, signer, onAfterSet)
```

Comment:
Might be good to add comment to explain as`mapSet` is generic name

Reply:
https://github.com/zeta-chain/node/pull/2497/commits/2f06e5639910af4fbfeb4261ff0faf4a86b53af1